tests/ui/cast.rs

```diff
@@ -61,4 +61,15 @@ fn main() {
     1u32 as usize; // Should not trigger any lint
     1i32 as isize; // Neither should this
     1i32 as usize;
+
+    // Test cast_unnecessary
+    1i32 as i32;
+    1f32 as f32;
+    false as bool;
+    &1i32 as &i32;
+
+    1i32 as i64;      // Should not trigger
```

Comment:
Good point. Current implementation will still lint for this.
But I've got no idea how to detect this since the type of the literal number is always the same as the casting target. e.g. `1 as u64` will have `1` with type `u64` and `1 as i32` will have `1` with type `i32`.
Any suggestion?